src/figuresClasses.ts

```diff
@@ -1,11 +1,82 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const biggestSide = Math.max(a, b, c);
+    const arr = [a, b, c];
+    let count = 0;
+
+    for (const item of arr) {
+      if (item !== biggestSide) {
+        count += item;
+      }
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0 || biggestSide >= count) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+      ) / 100
+    );
```

Comment:
[CODE STYLE] - The `getArea` function should not accept any parameters as all necessary values are available as object fields. Please remove the parameters and use the class properties directly.